src/main/scala/ignition/core/cache/ExpiringMultiLevelCache.scala

```diff
@@ -128,94 +143,117 @@ case class ExpiringMultiLevelCache[V](ttl: FiniteDuration,
 
   protected def now = DateTime.now.withZone(DateTimeZone.UTC)
 
-  private def timestamp(v: V) = TimestampedValue(now, v)
+  private def timestamp(v: V) = TimestampedValue(date = now, value = Some(v))
+
+  private def timestamp(hasError: Boolean, throwable: Throwable): TimestampedValue[V] = {
+    TimestampedValue(value = None, date = now, hasError = hasError, throwable = Some(throwable))
+  }
 
   private def elapsedTime(startNanoTime: Long) = FiniteDuration(System.nanoTime() - startNanoTime, TimeUnit.NANOSECONDS)
 
   private def remoteLockKey(key: Any) = s"$key-emlc-lock"
 
+  private def checkSavedErrorCache(key: String, genValue: () => Future[V], startTime: Long, v: TimestampedValue[V]): Future[V] = {
```

Comment:
Why this method exists?
The only thing it does is convert an Option[Throwable] in a Future[V].

```scala
v.throwable.map(Future.failed[V]).orElse(Future.successful(v.value))
```